diesel_cli/src/main.rs

```diff
@@ -172,19 +172,37 @@ fn migrations_dir(matches: &ArgMatches) -> PathBuf {
 }
 
 fn run_setup_command(matches: &ArgMatches) {
-    migrations::find_migrations_directory()
-        .unwrap_or_else(|_| {
-            create_migrations_directory()
-                .unwrap_or_else(handle_error)
+    let migrations_dir = create_migrations_dir(matches).unwrap_or_else(handle_error);
+
+    database::setup_database(matches, &migrations_dir).unwrap_or_else(handle_error);
+}
+
+fn create_migrations_dir(matches: &ArgMatches) -> DatabaseResult<PathBuf> {
+    let dir = matches.value_of("MIGRATION_DIRECTORY")
+        .map(PathBuf::from)
+        .or_else(|| {
+            env::var("MIGRATION_DIRECTORY").map(PathBuf::from).ok()
+        }).unwrap_or_else(|| {
+            find_project_root().unwrap_or_else(handle_error).join("migrations")
         });
 
-    database::setup_database(matches).unwrap_or_else(handle_error);
+    if !dir.is_dir() {
```

Comment:
This seems like a weird condition to me. Why do we want to handle the case where the expected path exists but is a file/symlink rather than a directory? It seems like just checking whether it exists would be more clear.